psi4/src/psi4/libmints/basisset.cc

```diff
@@ -173,16 +173,35 @@ int BasisSet::n_ecp_core() const {
     return ncoreelectrons;
 }
 
+int BasisSet::_atom_to_period(int Z) {
+    if (Z > 118) {
+        throw PSIEXCEPTION("Atomic number beyond Oganesson");
+    }
+    static const std::vector<int> full_shell_values = {0, 2, 10, 18, 36, 54, 86, 118};
+    auto period = std::lower_bound(full_shell_values.begin(), full_shell_values.end(), Z) - full_shell_values.begin();
+    return period;
+}
+
+int BasisSet::_period_to_full_shell(int p) {
+    if (p > 7) {
+        throw PSIEXCEPTION("Atomic number beyond Oganesson");
+    }
+    static const std::vector<int> full_shell_values = {0, 2, 10, 18, 36, 54, 86, 118};
+    return full_shell_values[p];
+}
+
 int BasisSet::n_frozen_core(const std::string &depth, SharedMolecule mol) {
     std::string local = depth;
     if (depth.empty()) local = Process::environment.options.get_str("FREEZE_CORE");
 
     SharedMolecule mymol = mol ? mol : molecule_;
 
-    if (local == "FALSE") {
+    if (local == "FALSE" or local == "0") {
         return 0;
-    } else if (local == "TRUE") {
-        int nfzc = 0;
+    } else if (local == "TRUE" or local == "1") {
```

Comment:
Could there be a problem if the user specifies "1" intending to freeze only 1 orbital, but this line interprets it as true?

Reply:
It'd be a weird corner case, admittedly.  I need more coffee to be able to think it through

Reply:
I think you're looking for `FROZEN_DOCC int` to freeze only `int` orbitals.

Reply:
You're completely correct.  I just got down to the `read_options` and upon seeing the allowed values I realized how stupid my comment was.  Coffee time!